File: lib/models/medico.dart
class Medico {
  final int medico_id;
  final int usuario_id;
  final String especialidad;
  final String centro_medico;

  Medico({
    required this.medico_id,
    required this.usuario_id,
    required this.especialidad,
    required this.centro_medico,
  });

  factory Medico.fromJson(Map<String, dynamic> json) {
    return Medico(
      medico_id: json['medico_id'] ?? 0,
      usuario_id: json['usuario_id'] ?? 0,
      especialidad: json['especialidad'] ?? '',
      centro_medico: json['centro_medico'] ?? '',
    );
  }
}

File: android/app/src/main/kotlin/com/example/mi_app/database/Medico.kt
package com.example.mi_app.database

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "Medico",
    foreignKeys = [ForeignKey(
        entity = Usuario::class,
        parentColumns = ["usuario_id"],
        childColumns = ["usuario_id"],
        onDelete = ForeignKey.CASCADE
    )]
)
data class Medico(
    @PrimaryKey(autoGenerate = true)
    val medico_id: Int = 0,
    val usuario_id: Int,
    val especialidad: String,
    val centro_medico: String,
    val nro_colegiatura: String
)
